core/arch/asm_defines.asm

```diff
@@ -563,6 +563,7 @@ ASSUME fs:_DATA @N@\
 # define ARG6 REG_R15
 # define ARG7 REG_R16
 # define ARG8 REG_R17
+# define SYSNUM_REG REG_R17
```

Comment:
Based on https://man7.org/linux/man-pages/man2/syscall.2.html a7 holds the system call number.

Reply:
You are right. But a7 is the abi name, a7 equal x17. Based on https://github.com/DynamoRIO/dynamorio/blob/master/core/arch/asm_defines.asm#L399 and https://github.com/DynamoRIO/dynamorio/blob/master/core/arch/asm_defines.asm#L565. REG_R17 exactly is a7.